File: frontend/src/types/review.ts
export interface Review {
    goodsId: string;
    userId: string;
    userName: string;
    rating: number;
    comment: string;
    createdAt: string;
    updatedAt: string;
}

export interface CreateReviewRequest {
    goodsId: string;
    rating: number;
    comment: string;
}

export interface GoodsRating {
    averageRating: number;
    totalReviews: number;
}

File: backend/src/main/kotlin/com/example/webflux/domain/model/Review.kt
package com.example.webflux.domain.model

import java.time.Instant

data class Review(
    val id: String,
    val goodsId: String,
    val userId: Long,
    val userName: String,
    val rating: Int,
    val comment: String,
    val createdAt: Instant,
    val updatedAt: Instant
)
